addons/website_sale/models/res_partner.py

```diff
@@ -18,8 +18,31 @@ def _compute_last_website_so_id(self):
             if website and not is_public:
                 partner.last_website_so_id = SaleOrder.search([
                     ('partner_id', '=', partner.id),
+                    ('pricelist_id', '=', partner.property_product_pricelist.id),
```

Comment:
did you check the query count of this search vs the previous one ? 

It is a function called a lot of time

Reply:
Quick test:

- I'm getting three extra queries when running cold (381 -> 384 for `/shop/cart`)
- One extra query when running hot (242 -> 243 for `/shop/cart`)

Reply:
Something seems weird with those numbers, hot state is supposed to be way less requests than cold one, something like 40-100 (if is ~400 on hot).
Not that it sometimes takes 2-3 reload to actually really be on hot state.
But maybe this is correct..

Reply:
Just to be sure, do we want to prevent the user to get his last cart back if that's not the pricelist we set for him?
I thought the PR was just about notifying the salesman than the change might not be reflected to his customer
(I was just thinking that at that point, if we are going to prevent the last SO retrieval if there is a pl mismatch, we might as well just change the pricelist of his open/draft SO (we already found it in the onchange)

Reply:
yes, as discuss it is expected by 'spec'.

I keep conversation open for hot / cold

Don't forget that you have several workers, and so several cache when you do perf test... as explain by rde

Reply:
So yes, the numbers stay the same. The cold run is a little different from one time to the next, but on average three extra queries. The hot run stays at the same level, even after multiple repeats.

Reply:
On my end it doesn't seems to change anything, which makes sense as this property should be cached.

So this looks like a safe change.

With only `website_sale`, logged as admin with one product in cart:

<table>
   <tr>
      <td/>
      <td>SQL Queries (cold)</td>
      <td>SQL Queries (hot)</td>
   </tr>
   <tr>
      <td colspan="3"> /shop/cart</td>
   </tr>
   <tr>
      <td> Before </td>
      <td>               388</td>
      <td>         83 </td>
   </tr>
   <tr>
      <td>After</td>
      <td>              388</td>
      <td>          83 </td>
         </tr>
   <tr>
      <td colspan="3"> /shop
   </tr>
   <tr>
      <td> Before  </td>
      <td>                  506 </td>
      <td>             191 </td>
   </tr>
   <tr>
       <td>After</td>
      <td>                 506  </td>
      <td>           191</td>  
   </tr>
</table>